move/sources/refund.move

```diff
@@ -0,0 +1,429 @@
+        } else {
+            none()
+        }
+    }
+    
+    public fun amount_to_claim_boosted(pool: &RefundPool, claimer: address): Option<u64> {
+        if (table::contains(&pool.unclaimed, claimer)) {
+            let claim_amount = *table::borrow(&pool.unclaimed, claimer);
+            some(claim_amount + div(claim_amount, 2))
+        } else {
+            none()
+        }
+    }
+    
+    public fun get_total_to_refund(pool: &RefundPool): u64 { total_to_refund(&pool.accounting) }
+    public fun get_total_raised(pool: &RefundPool): u64 { total_raised(&pool.accounting) }
+    public fun get_total_refunded(pool: &RefundPool): u64 { total_refunded(&pool.accounting) }
+    public fun get_total_boosted(pool: &RefundPool): u64 { total_boosted(&pool.accounting) }
+    public fun base_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.base_pool)) }
+    public fun booster_funds(pool: &RefundPool): u64 { balance::value(funds(&pool.booster_pool)) }
+    public fun current_liabilities(pool: &RefundPool): u64 { accounting::current_liabilities(&pool.accounting) }
+
+    // === Friends ===
+
+    public(friend) fun uid_mut(pool: &mut RefundPool): &mut UID { &mut pool.id }
+
+    public(friend) fun unclaimed_mut(pool: &mut RefundPool): &mut Table<address, u64> { &mut pool.unclaimed }
+    public(friend) fun accounting_mut(pool: &mut RefundPool): &mut Accounting { &mut pool.accounting }
+    public(friend) fun booster_pool_mut(pool: &mut RefundPool): &mut Pool { &mut pool.base_pool }
+
+    public(friend) fun reclaim_funds_(
+        inner_pool: &mut Pool,
+        total_raised: u64,
+        ctx: &mut TxContext,
+    ) {
+        let funders = funders_mut(inner_pool);
+        assert!(table::contains(funders, sender(ctx)), EInvalidFunder);
+        let funding_amount = table::remove(funders, sender(ctx));
+
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(inner_pool);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            let leftovers = balance::value(funds);
+
+            // ReclaimAmount = Leftovers * % Share <=>
+            // ReclaimAmount = Leftovers * FundingAmount/TotalRaised
+            // 
+            // We first upscale then downscale
+            div(
+                mul(leftovers, funding_amount),
+                total_raised
+            )
+        };
+
+        let reclaim_funds = coin::from_balance(balance::split(funds, reclaim_amount), ctx);
+        transfer::public_transfer(reclaim_funds, sender(ctx));
+    }
+
+    // === Phase Transitions ===
+    
+    public entry fun start_funding_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        timeout_ts: u64,
+        clock: &Clock,
+    ) {
+        assert_publisher(pub);
+        assert_address_addition_phase(pool);
+        
+        assert!(timeout_ts > clock::timestamp_ms(clock), EInvalidTimeoutTimestamp);
```

Comment:
Would be great to have a minimum default period.

Reply:
`assert!(timeout_ts >= clock::timestamp_ms(clock) + MIN_DEFAULT_PERIOD, EInvalidTimeoutTimestamp);`